pkg/v1/remote/index.go

```diff
@@ -182,3 +182,49 @@ func (r *remoteIndex) childDescriptor(child v1.Descriptor, platform v1.Platform)
 		platform:   platform,
 	}, nil
 }
+
+// matchesPlatform checks if the given platform matches the required platforms.
+// The given platform matches the required platform if
+// - architecture and OS are identical
+// - OS version and variant are identical if provided
+// - features and OS features of the required platform are subsets of those of the given platform
+func matchesPlatform(given, required v1.Platform) bool {
```

Comment:
Want to write a unit test for `matchesPlatform`?

Reply:
This is a good candidate for some simple [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests) since it's a pure function.

We have some similar ones [here](https://github.com/google/go-containerregistry/blob/2687bd5ba651f247a89696f9dc8120100a12f763/pkg/name/registry_test.go#L139-L215).